# Metagenome Hybrid Assembly with Cerulean
by Tessa Rhinehart (contact: tessa.rhinehart at gmail dot com)

This notebook walks through the steps used to create a metagenomic assembly using Cerulean software loaded on an 
Amazon Web Services machine image (AMI).


# Getting started
Use the CeruleanTools AMI and attach the Pink Berries data volume. When your instance boots up, mount the volume, e.g. using the following commands to inspect the volumes and mount one to a folder called `~/data`.
```
lsblk
mkdir ~/data
sudo mount /dev/xvdf/ ~/data
```

If you make a snapshot in the right zone (e.g. us-east-1d), you can load and mount the volume directly from your instance:

```
mkdir ~/data

aws ec2 attach-volume --volume-id vol-0bdfad3677d717075 --instance-id i-0a62227ff1d1977bd --device /dev/xvdh

sudo mount /dev/xvdh ~/data
```


# Pre-processing

The relevant data for metagenomic reads originated from one sample, and were generated by two different sequencing platforms. The long reads are uncorrected PacBio reads, and the short reads are Illumina. Respectively, these files are located at:

```
~/data/metagenomes/sequence_reads/pacbio/filtered_subreads.fastq
~/data/metagenomes/sequence_reads/illumina_4pacbio/Lizzy9.fastq
```

### Deinterleaving

These are paired-end reads 
